python/whylogs/api/logger/experimental/multi_dataset_logger/multi_dataset_rolling_logger.py

```diff
@@ -49,7 +49,7 @@ def __init__(self, dataset_timestamp: int, schema: Optional[DatasetSchema]) -> N
             self._target = DatasetProfile(dataset_timestamp=self._dataset_timestamp)
 
     def _has_segments(self) -> bool:
-        return self._schema is not None and self._schema.segments is not None
+        return self._schema is not None and bool(self._schema.segments)
```

Comment:
The explicit bool is not needed: https://google.github.io/styleguide/pyguide.html#214-truefalse-evaluations

The check for not None is ok, but you can simplify to:
`return self._schema and self._schema.segments`

Reply:
mypy doesn't like it if I remove the bool(). It infers the function as trying to return `Dict | None` instead of a boolean.

Reply:
In fact, I need the `is not None` as well it seems

Reply:
I see you are correct with respect to matching that return type of `bool` and maintaining one liner simplicity.